### PR TITLE
fix: activity label zero keys

### PR DESCRIPTION
**Pull request by huglx**
- do not show activities with an empty list of modified entities

### CODE REVIEW

File: backend/data/src/main/kotlin/io/tolgee/repository/activity/ActivityRevisionRepository.kt [hunk 1]
```diff
@@ -14,6 +14,7 @@ interface ActivityRevisionRepository : JpaRepository<ActivityRevision, Long> {
     """
     from ActivityRevision ar
     where ar.projectId = :projectId and ar.type is not null and ar.batchJobChunkExecution is null and ar.type in :types
+    and ar.modifiedEntities is not empty
```
Comment (JanCizmar) [inline backend/data/src/main/kotlin/io/tolgee/repository/activity/ActivityRevisionRepository.kt:17]: There is `saveWithoutModification` property in `ActivityType` `io/tolgee/activity/data/ActivityType.kt:21`. It's not used, but it's prepared there, so potentially, there can be activities in the list without any modified entity. I believe, this activity should have never been stored and the question is why it's being stored in here. `io.tolgee.activity.ActivityService#storeActivityData`

Comment (JanCizmar) [reply backend/data/src/main/kotlin/io/tolgee/repository/activity/ActivityRevisionRepository.kt:17]: Maybe you can fix this by not storing the activity data when (modifiedEntities.isEmpty() && !actvityType.saveWithoutModification).

Comment (JanCizmar) [reply backend/data/src/main/kotlin/io/tolgee/repository/activity/ActivityRevisionRepository.kt:17]: In the same time I would fix the UI, rather then fixing the backend for this particular case.

**Review by JanCizmar — CHANGES_REQUESTED** [review]

**Comment by huglx** [comment]
@JanCizmar great! I've already added a test and your code passed it, so we can merge I guess.

**Review by JanCizmar — APPROVED** [review]

**Review by JanCizmar — APPROVED** [review]

**Pull request merged**